### PR TITLE
Chore: Server API types POC - loadMessagesForRoom

### PR DESCRIPTION
**Pull request by GleidsonDaniel**
## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Improvement (non-breaking change which improves a current function)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation update (if none of the other choices apply)

## Checklist

- [ ] I have read the [CONTRIBUTING](https://github.com/RocketChat/Rocket.Chat/blob/develop/.github/CONTRIBUTING.md#contributing-to-rocketchat) doc
- [ ] I have signed the [CLA](https://cla-assistant.io/RocketChat/Rocket.Chat.ReactNative)
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works (if applicable)
- [ ] I have added necessary documentation (if applicable)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: app/lib/methods/updateMessages.ts [hunk 1]
```diff
@@ -49,7 +49,7 @@ export default async function updateMessages({
 			.query(Q.where('subscription_id', rid), Q.where('id', Q.oneOf(messagesIds)))
 			.fetch();
 
-		update = update.map(m => buildMessage(m));
+		update = update.map(m => buildMessage(m)) as TMessageModel[];
```
Comment (diegolmello) [inline app/lib/methods/updateMessages.ts:52]:
```suggestion
		update = update.map(m => buildMessage(m)) as IMessage[];
```

File: app/lib/methods/loadMessagesForRoom.ts [hunk 2]
```diff
@@ -36,25 +35,27 @@ export default function loadMessagesForRoom(args: {
...
 						msg: lastMessage.msg
 					};
-					data.push(loadMoreItem);
+					loadMoreMessages.push(loadMoreMessage);
 				}
-				await updateMessages({ rid: args.rid, update: data, loaderItem: args.loaderItem });
-				return resolve(data);
+				// @ts-ignore - How does the update work if the decrypt gives deconstruct on prop that doesn't exist?
+				await updateMessages({ rid: args.rid, update: [...data, ...loadMoreMessages], loaderItem: args.loaderItem });
+				return resolve([...data, ...loadMoreMessages]);
```
Comment (diegolmello) [inline app/lib/methods/loadMessagesForRoom.ts:58]: How about this?
```ts
const data: (IMessageFromServer | ILoadMoreMessage)[] = await load(args);
if (data?.length) {
	// const loadMoreMessages: ILoadMoreMessage[] = [];
	const lastMessage = data[data.length - 1];
	const lastMessageRecord = await getMessageById(lastMessage._id);
	if (!lastMessageRecord && data.length === COUNT) {
		const loadMoreMessage: ILoadMoreMessage = {
			_id: generateLoadMoreId(lastMessage._id),
			rid: lastMessage.rid,
			ts: moment(lastMessage.ts).subtract(1, 'millisecond').toString(),
			t: MESSAGE_TYPE_LOAD_MORE,
			msg: lastMessage.msg
		};
		data.push(loadMoreMessage);
	}
	// @ts-ignore - How does the update work if the decrypt gives deconstruct on prop that doesn't exist?
	await updateMessages({ rid: args.rid, update: data, loaderItem: args.loaderItem });
	return resolve(data);
```

We're going to be building the array with both types anyway.

File: app/definitions/rest/v1/history.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
+import type { IMessageFromServer } from '../../IMessage';
+
+export type HistoryEndpoints = {
```
Comment (diegolmello) [inline app/definitions/rest/v1/history.ts:3]: Endpoints are `${section}.*`, so they should be on `./v1/channels.ts`, `./v1/im.ts`, `./v1/groups.ts`

**Review by diegolmello — CHANGES_REQUESTED** [review]

File: app/lib/methods/loadMessagesForRoom.ts (hunk 2 above)
Comment (GleidsonDaniel) [reply app/lib/methods/loadMessagesForRoom.ts:58]: Good idea

**Review by diegolmello — APPROVED** [review]

**Pull request merged**